### DISCUSSION TITLE
Unable to extract data from table if it in columns

### DISCUSSION THREAD

**Discussion by danny007in** [pymupdf/PyMuPDF#2831] [Q&A]
[BULL_ENGG_FREE_DIP_2023english.pdf](https://github.com/pymupdf/PyMuPDF/files/13432704/BULL_ENGG_FREE_DIP_2023english.pdf)

Unable get table data ? by find_tables
Any idea

![image](https://github.com/pymupdf/PyMuPDF/assets/50356015/9ff781ac-34c6-47f7-9dc8-8901de4bf555)

**Comment by JorjMcKie** [comment]
This is no bug, but a Discussions post. Let me transfer first.

**Reply by JorjMcKie** [reply]
You have not specified your PyMuPDF version. The recent version does detect the tables - although it takes a **_very long time_** to execute `page.find_tables()`.
No surprise though b/o the enormous size of the page (1224 x 1008) and the amount of data.
Anyway, there are 64 tables correctly detected, here is a partial screen print:

![image](https://github.com/pymupdf/PyMuPDF/assets/8290722/590f2138-7781-47df-a992-71a93e98feb1)

How to make sense of these data remains a challenging task, and I am afraid there is nothing that can help you doing this ...

**Reply by danny007in** [reply]
I noticed that SNQ and TOT is not detected

**Comment by JorjMcKie** [comment]
> I noticed that SNQ and TOT is not detected

At least TOT rightly so: it has its own surrounding rectangle, and no connection to anything else. You must be a human to **_guess_** what it may mean (from the text, semantically interpreting it as meaning "total").
Impossible for any algorithm to realize - except maybe advanced AI, which this table finding definitely is not.
And a 1 x 1 table is never accepted as a table.

I am not sure about "SN\nQ" though.
